src/components/NormalRightRow/index.jsx

```diff
@@ -153,6 +145,24 @@ NormalRightRow.propTypes = {
     row: string.isRequired,
     line: string.isRequired,
   }).isRequired,
+  /**
+   * Tree node object data structure.
+   */
+  treeNode: shape({
+    /**
+     * Schema given to render upon. May also be a sub-schema in case
+     * for array items, object properties or more complex schemas.
+     */
+    schema: shape({
+      /** Type of schema or sub-schema */
+      type: string,
+      /** Name of schema or sub-schema */
+      name: string,
+    }).isRequired,
+    path: arrayOf(number),
+    /** children nodes of the current node */
+    children: array,
+  }).isRequired,
```

Comment:
It looks like we're using this prop-type in both `NormalRightRow` and `NormalLeftRow`. In the case of change, it will probably be best if we only need to change it in one place. We could start a `prop-types.js` and refer to that file instead.

```jsx
import { ... } from 'prop-types';
import { treeNode } from 'path/to/our/prop-types/file';

/**
* Tree node object data structure
*/
treeNode,
```

Reply:
Great idea! Applied changes!